sentry_sdk/serializer.py

```diff
@@ -98,6 +117,7 @@ def serialize(event, **kwargs):
     memo = Memo()
     path = []  # type: List[Segment]
     meta_stack = []  # type: List[Dict[str, Any]]
+    span_description_bytes = []
```

Comment:
I think this can be a simple integer, that would probably lower memory usage a tiny little bit. Since we don't have `nonlocal` you can still use a fixed-length list like this:

```
span_description_bytes[0] += value
```

instead of:

```
span_description_bytes.append(value)
```

Reply:
I've considered both; `nonlocal` is not an option in Python 2.7 and among list with single int or adhoc class with mutable field I thought list of lengths is the least surprising option. This piece of code is already super hard to read and understand, a list of ~999 ints vs list of one int makes no practical difference.

Reply:
I get the argument about readability but this is an extremely hot code path. Still you're right that this is unlikely to regress perf